### PR TITLE
feat: Migrate < Link />, < StoryDescription /> to ts

### LINKED ISSUE #76: TS migration: < Link />, < StoryDescription />
[Monday ticket](https://monday.monday.com/boards/3532714909/pulses/5336788875)

[Typescript migration guideline](https://github.com/mondaycom/vibe-storybook-components/blob/master/TYPESCRIPT_MIGRATION.md)

### PR DESCRIPTION
**Pull request by Dhoni77**
Closes #76

- Migrate Link and StoryDescription components to ts

### CODE REVIEW

File: src/components/link/link.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { FC } from 'react';
+import cx from 'classnames';
+import CoreLink from '../../helpers/components/Link/Link';
+import styles from './link.module.scss';
+import { Size } from './LinkConstants';
+
+type LinkProps = {
+  className: string;
```
Comment (SergeyRoyt) [inline src/components/link/link.tsx:8]:
```suggestion
  className?: string;
```

File: src/components/link/link.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  children: string;
+  href: string;
+  size?: Size;
+  withoutSpacing?: boolean;
+};
+
+const Link: FC<LinkProps> = ({ children, href, size = Size.MEDIUM, withoutSpacing = false, className }) => (
```
Comment (SergeyRoyt) [inline src/components/link/link.tsx:15]:
```suggestion
const Link: FC<LinkProps> & { sizes?: typeof Size } = ({ children, href, size = Size.MEDIUM, withoutSpacing = false, className }) => (
```

File: src/components/link/link.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    href={href}
+    className={cx(styles.compsLink, className, {
+      [styles.small]: size === Size.SMALL,
+      [styles.medium]: size === Size.MEDIUM,
+      [styles.withSpacing]: !withoutSpacing,
+    })}
+  />
+);
+
+export default Link;
```
Comment (SergeyRoyt) [inline src/components/link/link.tsx:27]:
```suggestion
export default withStaticProps(Link, { sizes: Size } );
```

So we'd be able to use it like this e.g. in other packages
`<Link size={Link.sizes.MEDIUM}`

Same as we do in `Tip`

File: src/components/link/LinkConstants.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export enum Size {
```
Comment (SergeyRoyt) [inline src/components/link/LinkConstants.ts:1]:
```suggestion
export enum LinkSize {
```

File: src/components/story-description/story-description.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import React, { FC, useMemo } from 'react';
+import cx from 'classnames';
+import styles from './story-description.module.scss';
+import { ElementContent } from '../../types';
+import { FlexAlign, FlexDirection, FlexGap, FlexJustify } from 'src/helpers/components/Flex/FlexConstants';
+import Flex from 'src/helpers/components/Flex/Flex';
+
+type StoryDescriptionProps = {
+  align?: FlexAlign;
+  description?: string | ElementContent;
```
Comment (SergeyRoyt) [inline src/components/story-description/story-description.tsx:10]:
```suggestion
  description?: ElementContent;
```
Already contains string

File: src/components/story-description/story-description.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import React, { FC, useMemo } from 'react';
+import cx from 'classnames';
+import styles from './story-description.module.scss';
+import { ElementContent } from '../../types';
+import { FlexAlign, FlexDirection, FlexGap, FlexJustify } from 'src/helpers/components/Flex/FlexConstants';
+import Flex from 'src/helpers/components/Flex/Flex';
+
+type StoryDescriptionProps = {
+  align?: FlexAlign;
```
Comment (SergeyRoyt) [inline src/components/story-description/story-description.tsx:9]: Let's also add this enum to static props, so we'd be able to reference it like this `StoryDescription.align.START`

File: src/components/story-description/story-description.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  description?: string | ElementContent;
+  children: ElementContent;
+  className: string;
+  headerAlign?: FlexAlign;
+  headerJustify?: FlexJustify;
```
Comment (SergeyRoyt) [inline src/components/story-description/story-description.tsx:14]: Let's add this enums to static too

File: src/components/story-description/story-description.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  headerStyle: Record<string, unknown>;
+  justify?: FlexJustify;
```
Comment (SergeyRoyt) [inline src/components/story-description/story-description.tsx:16]: Can also be a static prop

File: src/components/story-description/story-description.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+import Flex from 'src/helpers/components/Flex/Flex';
+
+type StoryDescriptionProps = {
+  align?: FlexAlign;
+  description?: string | ElementContent;
+  children: ElementContent;
+  className: string;
+  headerAlign?: FlexAlign;
+  headerJustify?: FlexJustify;
+  headerStyle: Record<string, unknown>;
```
Comment (SergeyRoyt) [inline src/components/story-description/story-description.tsx:15]:
```suggestion
  headerStyle: React.CSSProperties;
```

File: src/components/link/LinkConstants.ts [hunk 10]
```diff
@@ -0,0 +1,4 @@
...
+  SMALL = 'small',
+  MEDIUM = 'medium',
+};
```
Comment (SergeyRoyt) [inline src/components/link/LinkConstants.ts:4]: lint

**Review by SergeyRoyt — CHANGES_REQUESTED** [review]
Nice!
Let's just fix a few notes and make all checks green
Update master and merge from it to fix the build issues

**Comment by SergeyRoyt** [comment]
@Dhoni77  there are still some typescript issues to fix

**Comment by Dhoni77** [comment]
> @Dhoni77 there are still some typescript issues to fix

Hi @SergeyRoyt I ran lint I did not give any type errors

**Comment by SergeyRoyt** [comment]
> > @Dhoni77 there are still some typescript issues to fix
> 
> Hi @SergeyRoyt I ran lint I did not give any type errors

@Dhoni77 Yeah, but it's not a lint issue, it's a TS error
image
You have access to view details of these checks I ran, right?

**Comment by Dhoni77** [comment]
> > > @Dhoni77 there are still some typescript issues to fix
> > 
> > 
> > Hi @SergeyRoyt I ran lint I did not give any type errors
> 
> @Dhoni77 Yeah, but it's not a lint issue, it's a TS error image You have access to view details of these checks I ran, right?

Ok got it will fix this👍

**Comment by Dhoni77** [comment]
Hi @SergeyRoyt ,

In the `<Link />`  component should I make the href as optional property or send an empty string '' to `<Link />` in `<DeprecatedWarning />` to get rid of this typeerror.
Which one is expected ?

**Comment by SergeyRoyt** [comment]
> Hi @SergeyRoyt ,
> 
> In the `<Link />` component should I make the href as optional property or send an empty string '' to `<Link />` in `<DeprecatedWarning />` to get rid of this typeerror. Which one is expected ?

Hey, @Dhoni77 !
Let's probably make `alternativeLink` in the `DeprecatedWarning` not optional - that will fix the issue and makes sense

**Comment by Dhoni77** [comment]
cc @SergeyRoyt

**Review by SergeyRoyt — CHANGES_REQUESTED** [review]
Not sure where the lint issues are coming from, but, please, let's fix them before merge

**Review by SergeyRoyt — APPROVED** [review]
Nvm, will fix lint by myself
Thanks a lot!

**Pull request merged**